apps/worker/src/app/workflow/usecases/send-message/send-message-in-app.usecase.ts

```diff
@@ -65,6 +67,19 @@ export class SendMessageInApp extends SendMessageBase {
 
   @InstrumentUsecase()
   public async execute(command: SendMessageCommand) {
+    await this.executionDetailsArchiveProducer.add(
+      command.transactionId,
+      CreateExecutionDetailsCommand.create({
+        ...CreateExecutionDetailsCommand.getDetailsFromJob(command.job),
```

Comment:
Maybe in the future, we could define some property, let's call it for now `type`, and store only the minimalistic amount of data on the job, and then in the consumer based on the type we will do some logic, for ex. retrieve the job details. WDYT?

Reply:
I think it is a great idea and I believe we could add some strategies in order to build our execution details.